vraptor-core/src/main/java/br/com/caelum/vraptor/serialization/gson/GsonBuilderWrapper.java

```diff
@@ -90,9 +90,20 @@ private void registerAdapter(Class<?> adapterType, Object adapter) {
 	}
 	
 	private Class<?> getAdapterType(Object adapter) {
-		Type[] genericInterfaces = adapter.getClass().getGenericInterfaces();
-		ParameterizedType type = (ParameterizedType) genericInterfaces[0];
-		Type actualType = type.getActualTypeArguments()[0];
+		final Class<?> klazz;
```

Comment:
Instead of using String comparison you should use `CDIProxies.isCDIProxy` method to check if the class is a proxy.